test/e2e/common/autoscaling_utils.go

```diff
@@ -401,28 +400,14 @@ func (rc *ResourceConsumer) CleanUp() {
 	rc.stopWaitGroup.Wait()
 	// Wait some time to ensure all child goroutines are finished.
 	time.Sleep(10 * time.Second)
-	kind, err := kindOf(rc.kind)
-	framework.ExpectNoError(err)
+	kind := rc.kind.GroupKind()
 	framework.ExpectNoError(framework.DeleteResourceAndPods(rc.clientSet, rc.internalClientset, kind, rc.nsName, rc.name))
```

Comment:
`framework.DeleteResourceAndPods` calls `getRuntimeObjectForKind`, which only recognizes extensions.ReplicaSet and extensions.Deployment. Not sure whether you should add `apps` support to `getRuntimeObjectForKind` (would probably require other changes) or switch your types here to `extensions`

Reply:
Probably need to exercise these types in both the extensions and apps groups, so maybe both :-/